sampo/scheduler/lft/base.py

```diff
@@ -0,0 +1,218 @@
+import random
+import numpy as np
+from functools import partial
+from typing import Type, Iterable
+
+from sampo.scheduler.base import Scheduler, SchedulerType
+from sampo.scheduler.timeline.momentum_timeline import MomentumTimeline
+from sampo.schemas.time_estimator import WorkTimeEstimator, DefaultWorkEstimator
+from sampo.scheduler.lft.prioritization import lft_prioritization, lft_randomized_prioritization
+from sampo.scheduler.lft.time_computaion import work_duration
+
+from sampo.scheduler.timeline.base import Timeline
+from sampo.schemas.contractor import Contractor, get_worker_contractor_pool, WorkerContractorPool
+from sampo.schemas.graph import WorkGraph, GraphNode
+from sampo.schemas.landscape import LandscapeConfiguration
+from sampo.schemas.resources import Worker
+from sampo.schemas.schedule import Schedule
+from sampo.schemas.schedule_spec import ScheduleSpec
+from sampo.schemas.scheduled_work import ScheduledWork
+from sampo.schemas.time import Time
+from sampo.utilities.validation import validate_schedule
+
+from sampo.schemas.exceptions import IncorrectAmountOfWorker, NoSufficientContractorError
+
+
+class LFTScheduler(Scheduler):
```

Comment:
Why LFT not done on `GenericScheduler` base?

Reply:
Because LFTscheduler is based on different logic. It first assigns resources and then determines the order of works. If it will be based on GenericScheduler, then in addition to the resource_optimizer parameter, it will be also need to pass in parent's constructor None for the prioritization_f and optimize_resources_f parameters.
Also, if it will be based on GenericScheduler, then LFTScheduler will, in addition to the optimize_resources_using_spec method, have a get_default_res_opt_function method that has nothing to do with it.